runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -576,9 +576,32 @@ bool J9::ValuePropagation::transformUnsafeCopyMemoryCall(TR::Node *arraycopyNode
       TR::TreeTop *tt = _curTree;
       TR::Node *ttNode = tt->getNode();
 
+#if defined(J9VM_GC_ENABLE_SPARSE_HEAP_ALLOCATION)
+      if (TR::Compiler->om.isOffHeapAllocationEnabled()
+            && (ttNode->getOpCodeValue() == TR::treetop || ttNode->getOpCode().isResolveOrNullCheck()))
+         {
+         OMR::ValuePropagation *vp = this;
+         ListIterator<OMR::ValuePropagation::TR_TreeTopNodePair> copyMemoryIt(&vp->_offHeapCopyMemory);
+         bool alreadyAdded = false;
+         OMR::ValuePropagation::TR_TreeTopNodePair *copyMemoryPair;
+         for (copyMemoryPair = copyMemoryIt.getFirst();
+            copyMemoryPair; copyMemoryPair = copyMemoryIt.getNext())
+            {
+            if (copyMemoryPair->_node == arraycopyNode)
+               {
+               alreadyAdded = true;
+               break;
+               }
+            }
+
+         if (!alreadyAdded)
+            vp->_offHeapCopyMemory.add(new (vp->comp()->trStackMemory()) OMR::ValuePropagation::TR_TreeTopNodePair(tt, arraycopyNode));
```

Comment:
`_offHeapCopyMemory` doesn't seem to be processed in `doDelayedTransformations()` or to even be declared in the first place

Reply:
Added with https://github.com/eclipse/omr/pull/7391 in OMR

Reply:
Since that's `J9_PROJECT_SPECIFIC`, I'm guessing there isn't a particular reason for this to be in OMR instead of in OpenJ9. Could you please move that stuff into J9 instead? i.e. the list can be declared in `J9::ValuePropagation` and processed in `J9::ValuePropagation::doDelayedTransformations()`

Reply:
Hmm... why did I only see the OMR version of performing `doDelayedTransformations()`. Make sense.

Reply:
Fixed, need to revert the OMR change https://github.com/eclipse/omr/pull/7394